pkg/controller/core/admissioncheck_controller.go

```diff
@@ -0,0 +1,230 @@
+	r.watchers = append(r.watchers, watchers...)
+}
+
+func (r *AdmissionCheckReconciler) Create(e event.CreateEvent) bool {
+	ac, isAc := e.Object.(*kueue.AdmissionCheck)
+	if !isAc {
+		return false
+	}
+	defer r.notifyWatchers(nil, ac)
+	r.log.WithValues("admissionCheck", klog.KObj(ac)).V(5).Info("Create event")
+	if cqNames := r.cache.AddOrUpdateAdmissionCheck(ac); len(cqNames) > 0 {
+		r.qManager.QueueInadmissibleWorkloads(context.Background(), cqNames)
+	}
+	return true
+}
+
+func (r *AdmissionCheckReconciler) Update(e event.UpdateEvent) bool {
+	newAc, isAc := e.ObjectNew.(*kueue.AdmissionCheck)
+	if !isAc {
+		return false
+	}
+	oldAc, isAc := e.ObjectNew.(*kueue.AdmissionCheck)
+	if !isAc {
+		return false
+	}
+	defer r.notifyWatchers(oldAc, newAc)
+	r.log.WithValues("admissionCheck", klog.KObj(newAc)).V(5).Info("Update event")
+	if !newAc.DeletionTimestamp.IsZero() {
+		return true
+	}
+	if cqNames := r.cache.AddOrUpdateAdmissionCheck(newAc); len(cqNames) > 0 {
+		r.qManager.QueueInadmissibleWorkloads(context.Background(), cqNames)
+	}
+	return false
+}
+
+func (r *AdmissionCheckReconciler) Delete(e event.DeleteEvent) bool {
+	ac, isAc := e.Object.(*kueue.AdmissionCheck)
+
+	if !isAc {
+		return false
+	}
+	defer r.notifyWatchers(ac, nil)
+	r.log.WithValues("admissionCheck", klog.KObj(ac)).V(5).Info("Delete event")
+
+	if cqNames := r.cache.DeleteAdmissionCheck(ac); len(cqNames) > 0 {
+		r.qManager.QueueInadmissibleWorkloads(context.Background(), cqNames)
+	}
+	return true
+}
+
+func (r *AdmissionCheckReconciler) Generic(e event.GenericEvent) bool {
+	r.log.WithValues("object", klog.KObj(e.Object), "kind", e.Object.GetObjectKind().GroupVersionKind()).V(5).Info("Generic event")
+	return true
+}
+
+func (r *AdmissionCheckReconciler) NotifyClusterQueueUpdate(oldCq *kueue.ClusterQueue, newCq *kueue.ClusterQueue) {
+	log := r.log.WithValues("oldCq", klog.KObj(oldCq), "newCq", klog.KObj(newCq))
+	log.V(5).Info("Cluster queue notification")
+	noChange := newCq != nil && oldCq != nil && slices.CmpNoOrder(oldCq.Spec.AdmissionChecks, newCq.Spec.AdmissionChecks)
+	if noChange {
+		return
+	}
+
+	if oldCq != nil {
+		r.cqUpdateCh <- event.GenericEvent{Object: oldCq}
+	}
+
+	if newCq != nil {
+		r.cqUpdateCh <- event.GenericEvent{Object: newCq}
+	}
+}
+
+type acCqHandler struct {
+	cache *cache.Cache
+}
+
+func (h *acCqHandler) Create(context.Context, event.CreateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *acCqHandler) Update(context.Context, event.UpdateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *acCqHandler) Delete(context.Context, event.DeleteEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *acCqHandler) Generic(ctx context.Context, e event.GenericEvent, q workqueue.RateLimitingInterface) {
+	cq := e.Object.(*kueue.ClusterQueue)
+	if cq.Name == "" {
+		return
+	}
+	log := log.FromContext(ctx).WithValues("cq", klog.KObj(cq))
```

Comment:
```suggestion
	log := log.FromContext(ctx).WithValues("clusterQueue", klog.KObj(cq))
```

We shouldn't drop the context in logs. Let's keep using "clusterQueue" the same as other locations.